tests/unit/s2n_tls13_cert_verify_test.c

```diff
@@ -270,50 +302,126 @@ int run_tests(const struct s2n_tls13_cert_verify_test *test_case, s2n_mode verif
         EXPECT_SUCCESS(s2n_blob_init(&b, s2n_stuffer_raw_read(&certificate_out, available_size), available_size));
         if (verifying_conn->mode == S2N_CLIENT) {
             EXPECT_OK(s2n_asn1der_to_public_key_and_type(&verifying_conn->handshake_params.server_public_key, &pkey_type, &b));
+            EXPECT_SUCCESS(s2n_pkey_match(&verifying_conn->handshake_params.server_public_key, sending_conn->handshake_params.our_chain_and_key->private_key));
         } else {
             EXPECT_OK(s2n_asn1der_to_public_key_and_type(&verifying_conn->handshake_params.client_public_key, &pkey_type, &b));
+            EXPECT_SUCCESS(s2n_pkey_match(&verifying_conn->handshake_params.client_public_key, sending_conn->handshake_params.our_chain_and_key->private_key));
         }
 
         /* Hash initialization */
+        EXPECT_SUCCESS(s2n_hash_init(&sending_conn->handshake.hashes->sha256, S2N_HASH_SHA256));
+        EXPECT_SUCCESS(s2n_hash_update(&sending_conn->handshake.hashes->sha256, hello, strlen((char *) hello)));
         EXPECT_SUCCESS(s2n_hash_init(&verifying_conn->handshake.hashes->sha256, S2N_HASH_SHA256));
         EXPECT_SUCCESS(s2n_hash_update(&verifying_conn->handshake.hashes->sha256, hello, strlen((char *) hello)));
 
-        /* Send and receive with mismatched hash algs */
-        EXPECT_SUCCESS(s2n_tls13_cert_verify_send(verifying_conn));
-
-        /* Reinitialize hash */
-        EXPECT_SUCCESS(s2n_hash_init(&verifying_conn->handshake.hashes->sha256, S2N_HASH_SHA256));
-        EXPECT_SUCCESS(s2n_hash_update(&verifying_conn->handshake.hashes->sha256, hello, strlen((char *) hello)));
+        /* Send cert verify */
+        EXPECT_SUCCESS(s2n_tls13_cert_verify_send(sending_conn));
+        EXPECT_SUCCESS(s2n_stuffer_copy(&sending_conn->handshake.io, &verifying_conn->handshake.io, s2n_stuffer_data_available(&sending_conn->handshake.io)));
 
-        /* In this case it doesn't matter if we use server_cert_sig_scheme or client_cert_sig_scheme as they are currently equal */
         struct s2n_signature_scheme test_scheme = *verifying_conn->handshake_params.server_cert_sig_scheme;
         verifying_conn->handshake_params.server_cert_sig_scheme = &test_scheme;
+        verifying_conn->handshake_params.client_cert_sig_scheme = &test_scheme;
+
+        /* Use a hash algorithm different from sender */
         test_scheme.hash_alg = S2N_HASH_SHA1;
-        EXPECT_FAILURE(s2n_tls13_cert_read_and_verify_signature(verifying_conn,
-                verifying_conn->handshake_params.server_cert_sig_scheme));
 
-        /* send and receive with mismatched signature algs */
-        verifying_conn->handshake_params.client_cert_sig_scheme = &test_scheme;
-        test_scheme.hash_alg = S2N_HASH_SHA256;
-        test_scheme.sig_alg = S2N_SIGNATURE_ECDSA;
-        test_scheme.iana_value = 0xFFFF;
+        /* Skip the first two bytes (signature scheme) and start reading from signature size */
```

Comment:
What we want here is a comment explaining _why_ this skip_read is necessary. Why do you need the read cursor to be moved?